Rubberduck.VBEEditor/QualifiedModuleName.cs

```diff
@@ -33,20 +33,43 @@ public static string GetProjectId(IReference reference)
             return new QualifiedModuleName(projectName, reference.FullPath, projectName).ProjectId;
         }
 
+        public static int GetModuleContentHash(IVBComponent component)
+        {
+            if (component == null || component.IsWrappingNullReference)
+            {
+                return 0;
+            }
+
+            using (var codeModule = component.CodeModule)
+            {
+                return codeModule?.SimpleContentHash() ?? 0;
+            }
+        }
+
+
         public QualifiedModuleName(IVBProject project)
         {
             _componentName = null;
             ComponentType = ComponentType.Undefined;
             _projectName = project.Name;
             ProjectPath = string.Empty;
-            ProjectId = GetProjectId(project);           
+            ProjectId = GetProjectId(project);
+            ModuleContentHashOnCreation = GetModuleContentHash(null);
         }
 
         public QualifiedModuleName(IVBComponent component)
         {
             ComponentType = component.Type;
             _componentName = component.IsWrappingNullReference ? string.Empty : component.Name;
 
+            //note: We set this property in order to stabelize the component.
+            //For some reason, components sometimes seem to get removed on the COM side although 
+            //an RCW is still holding a reference. For some reason, opening the CodeModule of a 
+            //component seems to prevent this. 
+            //This is a hack to open the code module on each component for which we get a QMN 
+            //in a way that does not get optimized away.
```

Comment:
nice explanatory comment. Seems like a VBE bug to me TBQH. Has someone checked whether MS is aware of this?

Reply:
@Vogel612 I can ask around, but not expecting anything to be done on MSFT side of things... VBIDE API is pretty much carved in stone at this point.